indra/sources/tees/tees_api.py

```diff
@@ -104,6 +116,197 @@ def process_text(text, pmid=None, tees_path=None, python2_path=None):
+        Path to the TEES directory
+    python2_path: str
+        The path to the python 2 interpreter
+
+    Returns
+    -------
+    output_dir: str
+        Temporary directory with TEES output. The caller should delete this
+        directgory when done with it.
+    """
+    # Make sure the provided TEES directory exists
+    if not os.path.isdir(tees_path):
+        raise Exception('Provided TEES directory does not exist.')
+
+    # Make sure the classify.py script exists within this directory
+    classify_path = 'classify.py'
+    # if not os.path.isfile(classify_path):
+    #    raise Exception('classify.py does not exist in provided TEES path.')
+
+    # Create a temporary directory to tag the shared-task files
+    tmp_dir = tempfile.mkdtemp(suffix='indra_tees_processor')
+
+    pwd = os.path.abspath(os.getcwd())
+
+    try:
+        # Write text to a file in the temporary directory
+        text_path = os.path.join(tmp_dir, 'text.txt')
+        # Had some trouble with non-ascii characters. A possible TODO item in
+        # the future is to look into resolving this, for now just ignoring
+        # non-latin-1 characters
+        with codecs.open(text_path, 'w', encoding='latin-1', errors='ignore') \
+                as f:
+            f.write(text)
+
+        # Run TEES
+        output_path = os.path.join(tmp_dir, 'output')
+        model_path = os.path.join(tees_path, 'tees_data/models/GE11-test/')
+        command = [python2_path, classify_path, '-m', model_path,
+                   '-i', text_path,
+                   '-o', output_path]
+        try:
+            pwd = os.path.abspath(os.getcwd())
+            os.chdir(tees_path)  # Change to TEES directory
+            # print('cwd is:', os.getcwd())
+            # out = subprocess.check_output(command, stderr=subprocess.STDOUT)
+            p = subprocess.Popen(command, stdout=subprocess.PIPE,
+                                 stderr=subprocess.PIPE, cwd=tees_path)
+            p.wait()
+            (so, se) = p.communicate()
+            print(so)
+            print(se)
+            os.chdir(pwd)  # Change back to previous directory
+            # print('cwd is:', os.getcwd())
+            # print(out.decode('utf-8'))
+
+        except BaseException as e:
+            # If there's an error, print it out and then propagate the
+            # exception
+            os.chdir(pwd)  # Change back to previous directory
+            # print (e.output.decode('utf-8'))
+            raise e
+
+    except BaseException as e:
+        # If there was an exception, delete the temporary directory and
+        # pass on the exception
+        shutil.rmtree(tmp_dir)
+        raise e
+    # Return the temporary directory with the TEES output
+    output_tuple = extract_relevant_tees_output(tmp_dir)
+    shutil.rmtree(tmp_dir)
+    return output_tuple
+
+def extract_relevant_tees_output(output_dir):
+    """Extract the text of the a1, a2, and sentence segmentation files from the
```

Comment:
Why "relevant" in the name? If not essential, just rename to `extract_output` (as in previous comment `tees` is not needed in the name)